lib/async_hooks.js

```diff
@@ -365,21 +365,14 @@ function emitBeforeN(asyncId) {
 
 // Usage: emitBeforeS(asyncId[, triggerAsyncId]). If triggerAsyncId is omitted
 // then asyncId will be used instead.
-function emitBeforeS(asyncId, triggerAsyncId = asyncId) {
+function emitBeforeS(asyncId, triggerAsyncId) {
   // CHECK(Number.isSafeInteger(asyncId) && asyncId > 0)
   // CHECK(Number.isSafeInteger(triggerAsyncId) && triggerAsyncId > 0)
 
-  // Validate the ids.
-  if (asyncId < 0 || triggerAsyncId < 0) {
-    fatalError('before(): asyncId or triggerAsyncId is less than zero ' +
-               `(asyncId: ${asyncId}, triggerAsyncId: ${triggerAsyncId})`);
-  }
```

Comment:
This is to throw nicely. If the values make it to `AsyncWrap::PushAsyncIds()` then it will abort (i.e. always core dump).